### PR TITLE
workflow tab

### PR DESCRIPTION
**Pull request by maryhipp**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [x] Yes
- [ ] No, because:

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ ] No

## Description
* split the workflow editor into `Edit Mode` (what it is today) and `View Mode` (a view that only exposes the linear fields, if any, for the loaded workflow)
* adds an indicator to exposed fields if the values have changed from what the workflow defines as default, and gives the user an option to reset to the default views

## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] Yes
- [ ] No : _please replace this line with details on why tests
      have not been included_

### CODE REVIEW

File: invokeai/frontend/web/src/features/nodes/components/sidePanel/viewMode/WorkflowWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { useAppSelector } from '../../../../../app/store/storeHooks';
+import { useMemo } from 'react';
+
+export const WorkflowWarning = () => {
+  const nodesNeedUpdate = useGetNodesNeedUpdate();
+  const isTouched = useAppSelector((s) => s.workflow.isTouched);
+
+  const showWarning = useMemo(() => {
+    return nodesNeedUpdate || isTouched;
+  }, [nodesNeedUpdate, isTouched]);
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/nodes/components/sidePanel/viewMode/WorkflowWarning.tsx:15]: The dependencies here are already memoized primitive values and `showWarning` is only used once - wrapping them in a `useMemo` doesn't do anything. Can just `if (!nodeNeedsUpdate && !isTouched) { return null }`

**Comment by psychedelicious** [comment]
Tidied/fixed a few things.

Also fixed a longstanding bug where workflows were marked as touched when loading for the first time, or after clicking a node.

File: invokeai/frontend/web/src/features/nodes/store/workflowSlice.ts [hunk 2]
```diff
@@ -78,15 +95,43 @@ export const workflowSlice = createSlice({
...
+          value: input.value,
+        };
+        originalExposedFieldValues.push(originalExposedFieldValue);
+      });
+
+      return {
+        ...cloneDeep(initialWorkflowState),
+        ...cloneDeep(workflowExtra),
+        originalExposedFieldValues,
+        mode: state.mode,
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/nodes/store/workflowSlice.ts:133]: One potentially awkward UX is when you load a workflow without any exposed fields, you don't see anything in the options panel. 

If we wanted to default to edit mode in this situation, we could do this:
```suggestion
        mode: workflowExtra.exposedFields.length > 0 ? state.mode : 'edit',
```

But that's kinda confusing.

**Review by psychedelicious — APPROVED** [review]
- There are now 3 buttons to save a workflow in edit mode:
  - In the side panel
  - On the react flow canvas
  - In the `...` menu at top-right
  
  I don't think we need so many of them - maybe skip the side panel save button?

- I changed the mode toggle to be an `IconButton` w/ a tooltip.
  
  https://github.com/invoke-ai/InvokeAI/assets/4822129/9f41e126-5f79-4a50-a718-47d6a93cb490

- Also as you can see in the video, I changed the New Workflow icon to be different than the workflow tab icon. I also changed it in the `...` menu.

File: invokeai/frontend/web/src/features/nodes/store/workflowSlice.ts (hunk 2 above)
Comment (maryhipp) [reply invokeai/frontend/web/src/features/nodes/store/workflowSlice.ts:133]: Yeah agreed - I think lets leave it as-is for now

File: invokeai/frontend/web/src/features/nodes/store/types.ts [hunk 3]
```diff
@@ -33,9 +33,16 @@ export type NodesState = {
   selectionMode: SelectionMode;
 };
 
+export type WorkflowMode = 'edit' | 'view';
+export type FieldIdentifierWithValue = FieldIdentifier & {
+  value: StatefulFieldValue;
+};
```
Comment (maryhipp) [inline invokeai/frontend/web/src/features/nodes/store/types.ts:39]: 👍

**Pull request merged**